apps/meteor/app/slackbridge/server/SlackAdapter.js

```diff
@@ -41,7 +41,19 @@ export default class SlackAdapter {
 		}
 		this.slackAPI = new SlackAPI(this.apiToken);
 		this.rtm = new RTMClient(this.apiToken);
-		this.rtm.start();
+
+		Promise.await(
```

Comment:
is `Promise.await` needed?

Reply:
yes, if we don't use it, the `then()` and `catch()` doesn't get executed and we cannot handle the error.

Reply:
but you are still throwing the error, so if the application was breaking here before the changes its still breaking, or is there something I'm missing?

Reply:
Actually throwing the error and Adding Promise.await it will breaks the application for sure (thats why you were forced to add the try/catch here https://github.com/RocketChat/Rocket.Chat/pull/25793/files#diff-034e87d82cdd01feb879443b1f59f27a70dbebb6c8b60bc1827eec093942347dR34

Reply:
I think you can change the connect method to async and use `await` instead of `Promise.await` and then just handle the exception like that `slack.connect(apiToken).catch(e => connLogger.error('error connecting to slack', e)`

Reply:
changes done!